cli/src/commands/invoke/service/sqldb.rs

```diff
@@ -25,4 +40,211 @@ impl LocalSqlDB {
         // Be careful with password, change it in `DOCKER_COMPOSE_SNIPPET` accordingly
         "postgres://postgres:localdbpassword@localhost:5432/postgres?sslmode=disable".to_string()
     }
+
+    /// Provision database with a retry mechanism for handling connection issues
+    ///
+    /// If fixtures are provided, the method tries to connect to the database and load fixtures.
+    pub async fn provision(&self) -> eyre::Result<()> {
+        // Do nothing if there are no fixtures to load
+        if self.fixtures.is_empty() {
+            return Ok(());
+        }
+
+        let max_retries = 5;
+        let retry_delay_ms = 1000;
+
+        for attempt in 1..=max_retries {
+            // Retry only if the connection fails
+            let fixtures = Fixtures::new(self.connection_string(), &self.fixtures).await;
+
+            match fixtures {
+                Ok(fixtures) => {
+                    fixtures.apply().await.wrap_err("Failed to load fixtures")?;
+                    return Ok(());
+                }
+                Err(err) => {
+                    if attempt == max_retries {
+                        return Err(err);
+                    }
+
+                    tokio::time::sleep(std::time::Duration::from_millis(retry_delay_ms)).await;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// A structure representing SQL database fixtures used for setting up or interacting
+/// with a database during testing or other operations.
+struct Fixtures<'a> {
+    /// Connection pool to the database
+    connections_pool: Pool<Postgres>,
+
+    /// List of fixture files to load into the database
+    fixtures: &'a [String],
+}
+
+impl<'a> Fixtures<'a> {
+    pub async fn new(connection_string: String, fixtures: &'a [String]) -> eyre::Result<Self> {
+        Ok(Self {
+            connections_pool: PgPoolOptions::new()
+                .connect(&connection_string)
+                .await
+                .wrap_err("Filed to connect to database")?,
+
+            fixtures,
+        })
+    }
+
+    /// Apply all fixtures into the database
+    pub async fn apply(&self) -> eyre::Result<()> {
+        let current_dir = std::env::current_dir().wrap_err("Failed to get current directory")?;
+        let fixtures_dir = current_dir.join("fixtures");
```

Comment:
@flyin let's not force them keeping all fixtures in `fixtures/` dir. Let the path be a path relative to the current direction, don't add `fixtures/`.